x/subspaces/keeper/permissions.go

```diff
@@ -7,86 +7,107 @@ import (
 }
 
+// getSectionPermissions gets the permissions for the given user set inside the specified section only
+func (k Keeper) getSectionPermissions(ctx sdk.Context, subspaceID uint64, sectionID uint32, user string) types.Permission {
+	store := ctx.KVStore(k.storeKey)
+	return types.UnmarshalPermission(store.Get(types.UserPermissionStoreKey(subspaceID, sectionID, user)))
+}
+
 // GetUserPermissions returns the permissions that are currently set inside
 // the subspace with the given id for the given user
-func (k Keeper) GetUserPermissions(ctx sdk.Context, subspaceID uint64, user sdk.AccAddress) types.Permission {
-	store := ctx.KVStore(k.storeKey)
-	return types.UnmarshalPermission(store.Get(types.UserPermissionStoreKey(subspaceID, user)))
+func (k Keeper) GetUserPermissions(ctx sdk.Context, subspaceID uint64, sectionID uint32, user string) types.Permission {
+	if sectionID == 0 {
+		return k.getSectionPermissions(ctx, subspaceID, sectionID, user)
+	}
+
+	sectionPermissions := k.getSectionPermissions(ctx, subspaceID, sectionID, user)
+
+	// Get the parent permissions
+	parentPermissions := types.PermissionNothing
+	section, found := k.GetSection(ctx, subspaceID, sectionID)
+	if found {
+		parentPermissions = k.GetUserPermissions(ctx, subspaceID, section.ParentID, user)
+	}
+
+	return types.CombinePermissions(parentPermissions, sectionPermissions)
 }
 
 // GetGroupsInheritedPermissions returns the permissions that the specified user
 // has inherited from all the groups that they are part of.
-func (k Keeper) GetGroupsInheritedPermissions(ctx sdk.Context, subspaceID uint64, user sdk.AccAddress) types.Permission {
+func (k Keeper) GetGroupsInheritedPermissions(ctx sdk.Context, subspaceID uint64, sectionID uint32, user string) types.Permission {
 	var permissions []types.Permission
-	k.IterateSubspaceGroups(ctx, subspaceID, func(index int64, group types.UserGroup) (stop bool) {
-		if k.IsMemberOfGroup(ctx, subspaceID, group.ID, user) {
-			permissions = append(permissions, group.Permissions)
-		}
+
+	// Iterate over the section ancestors and get all the user groups for each ancestor
+	// to check if the user is part of a group
+	k.IterateSectionPath(ctx, subspaceID, sectionID, func(section types.Section) (stop bool) {
+		k.IterateSectionUserGroups(ctx, section.SubspaceID, section.ID, func(group types.UserGroup) (stop bool) {
+			if k.IsMemberOfGroup(ctx, subspaceID, group.ID, user) {
+				permissions = append(permissions, group.Permissions)
+			}
+			return false
+		})
 		return false
 	})
+
 	return types.CombinePermissions(permissions...)
 }
 
 // GetUsersWithPermission returns all the users that have a given permission inside the specified subspace
-func (k Keeper) GetUsersWithPermission(ctx sdk.Context, subspaceID uint64, permission types.Permission) ([]sdk.AccAddress, error) {
+func (k Keeper) GetUsersWithPermission(ctx sdk.Context, subspaceID uint64, permission types.Permission) ([]string, error) {
```

Comment:
I would name this function `GetUsersWithRootPermission`.

Reply:
Agree, done! Even though this method is only used inside simulation tests, `GetUsersWithRootPermission` makes more sense